### PR TITLE
Add IBM::MASS/MASSV targets

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Hard coded -lmass is problematic. Use rely on CMake for handling paths.

## What type(s) of changes does this code introduce?
- Build related changes

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Summit

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: CMake/FindIBMMASS.cmake [hunk 1]
```diff
@@ -2,37 +2,31 @@
...
+  MASS_LIBRARY mass
   HINTS ${MASS_ROOT}
   PATH_SUFFIXES ${SUFFIXES})
-if(NOT MASS_LINK_DIRECTORIES)
-  message(FATAL_ERROR "MASS_LINK_DIRECTORIES not set. ${MASS_FIND_LIB} " "not found in MASS_ROOT/(${SUFFIXES})")
-endif(NOT MASS_LINK_DIRECTORIES)
-message(STATUS "MASS_LINK_DIRECTORIES: ${MASS_LINK_DIRECTORIES}")
+message(STATUS "MASS_LIBRARY: ${MASS_LIBRARY}")
 
-set(MASS_LINKER_FLAGS -L${MASS_LINK_DIRECTORIES} -Wl,-rpath,${MASS_LINK_DIRECTORIES})
```
[the remark this answers is not in the document]
Comment (PDoakORNL) [reply CMake/FindIBMMASS.cmake:30]: But maybe we need to think about this in general and not just for mass.

File: CMake/FindIBMMASS.cmake [hunk 2]
```diff
@@ -2,37 +2,31 @@
 include(CheckCXXSourceCompiles)
 
 message(STATUS "Looking for IBM MASS libraries")
-# Finding and setting the MASS_INCLUDE_DIRECTORIES
-set(SUFFIXES include)
+
+# Finding MASS_INCLUDE
 find_path(
-  MASS_INCLUDE_DIRECTORIES name "mass.h"
```
Comment (PDoakORNL) [inline CMake/FindIBMMASS.cmake:8]: generally find modules set PACKAGE_INCLUDE_DIRECTORIES and PACKAGE_LIBRARIES.  However this shouldn't matter if we just use the targets from here on.

**Review by PDoakORNL — APPROVED** [review]
This may be a breaking change for some setups due to the removal of rpath

File: CMake/FindIBMMASS.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (ye-luo) [reply CMake/FindIBMMASS.cmake:30]: I have played with CMake on dynamic libraries.
1. find_library records AB_LIB=`/fullpath/libABC.so`
2. When AB_LIB is linked. CMake turns it into `-L/fullpath -lABC` and also add runpath `/fullpath`
So even if you unload the module, runpath will still find the library.

Additional note: RHEL or Centos turns `-Wl,-rpath` into rpath but other OS turn it into runpath. This doesn't really affect us but I'm in favor of runpath.

File: CMake/FindIBMMASS.cmake (hunk 2 above)
Comment (ye-luo) [reply CMake/FindIBMMASS.cmake:8]: Using target also simplify try_compiler. Quite nice to do it this way.

I found for include, I won't get much confused. We only need a path to the directory.
For library, I tend to make more strict choice mainly to distinguish the path the library and the library files themselves. LIBS or LIBRARIES must refer to the library files. `_DIRS` or ` _DIRECTORY` are needed for library directory.

**Comment by ye-luo** [comment]
Start testing in-house

**Pull request merged**